### PR TITLE
Issue/1883/dispatcher return values

### PR DESCRIPTION
**Pull request by john-parton**
Fix for https://github.com/django-oscar/django-oscar/issues/1883

The idea is to make `Dispatcher` more modular so that more elements can be customized. I'll be able to make some more useful additions after https://github.com/django-oscar/django-oscar/pull/1882 is merged.

I should probably write some tests. The `Dispatcher` class only has a few integration tests. Docs might be nice.

### CODE REVIEW

File: src/oscar/apps/customer/utils.py [hunk 1]
```diff
@@ -23,64 +23,75 @@ def __init__(self, logger=None):
...
+        Dispatch order-related messages to the customer.
         """
         if order.is_anonymous:
-            if 'email_address' in kwargs:
-                self.send_email_messages(kwargs['email_address'], messages)
-            elif order.guest_email:
-                self.send_email_messages(order.guest_email, messages)
-            else:
-                return
+            email = kwargs.get('email_address', order.guest_email)
```
Comment (mvantellingen) [inline src/oscar/apps/customer/utils.py]: To keep the same behaviour shouldn't we add `if not email: return`?

**Comment by mvantellingen** [comment]
Looks good @john-parton! One minor question added :-)

File: src/oscar/apps/customer/utils.py (hunk 1 above)
Comment (john-parton) [inline src/oscar/apps/customer/utils.py]: I'm not sure. I tried to keep the logic exactly the same.

There's a check in `send_user_email_messages`

Comment (sasha0) [inline src/oscar/apps/customer/utils.py]: I think early return is not necessary, since both methods have it.

Comment (john-parton) [inline src/oscar/apps/customer/utils.py]: I'm not totally sure what the question is. The behavior should be the same.

Maybe we need a test for the correct behavior?

Comment (sasha0) [inline src/oscar/apps/customer/utils.py]: Test will be useful in any case.

**Comment by sasha0** [comment]
Closing in favor of PR #2303, which is rebased version of this PR.

**Pull request closed without merging**